### PR TITLE
Kusk CLI `version` - Partially closes kubeshop/kusk-gateway#415

### PR DESCRIPTION
**Pull request by mbana**
Kusk CLI `version` - Partially closes kubeshop/kusk-gateway#415

Define a `version` command that uses value of `github.com/kubeshop/kusk-gateway/pkg/build` populated by `ldflags`:

* `github.com/kubeshop/kusk-gateway`: Get latest version that includes `github.com/kubeshop/kusk-gateway/pkg/build` package.
* `github.com/kubeshop/kusk-gateway/pkg/build`: `Version` variable will contain the latest tag when built.
* `TELEMETRY_TOKEN`: Remove references to `ldflags`/build-arg that is not used.
* `.github/workflows/go.yml`: Remove `test-GoReleaser-build` as we can just run it locally using `make build-goreleaser`.
* `.goreleaser.yml`: Populate `KuskGAMeasurementID` based on secret/env.
* `.goreleaser.yml`: Populate `KuskGAApiSecret` based on secret/env.

Example run log
---------------

```sh
$ make build
Makefile:62: warning: undefined variable 'GA_ID'
Makefile:62: warning: undefined variable 'GA_SECRET'
go build -v -o ./kusk -ldflags="-X 'github.com/kubeshop/kusk-gateway/pkg/build.Version=v1.0.4-5-ga63345e' -X 'github.com/kubeshop/kusk-gateway/pkg/analytics.KuskGAMeasurementID=' -X 'github.com/kubeshop/kusk-gateway/pkg/analytics.KuskGAApiSecret='" ./main.go
$ ./kusk version
kusk version 1.0.4-5-ga63345e
https://github.com/kubeshop/kusk/releases/latest
$ ./kusk --version
kusk version 1.0.4-5-ga63345e
https://github.com/kubeshop/kusk/releases/latest
$ ./kusk -v
kusk version 1.0.4-5-ga63345e
https://github.com/kubeshop/kusk/releases/latest
```

---

This PR...

## Changes

-

## Fixes

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Review by jasmingacic — CHANGES_REQUESTED** [review]
I recommend we add version at build time just like I did in kusk-gayeway repo.

If we pass it as a LDFLAG when building we can ensure it won't be changed. 

Correct me if I'm wrong?

**Comment by mbana** [comment]
> I recommend we add version at build time just like I did in kusk-gayeway repo.
> 
> If we pass it as a LDFLAG when building we can ensure it won't be changed.
> 
> Correct me if I'm wrong?

I took another approach which is to call `git` and use the `time` package to partially get the version information. If you don't like it, let me know, and I'll change it. Here's where to look: https://github.com/kubeshop/kusk/pull/37/files#diff-f2ae1f6d37751ba11f66da69f40d1c3076bb1f2246878641080d5855034881b3R27.

File: internal/build/build.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+func init() {
+	if Version == "DEV" {
+		if info, ok := debug.ReadBuildInfo(); ok && info.Main.Version != "(devel)" {
+			Version = info.Main.Version
+		}
+	}
+
+	if tag, err := cmdOutput("git", "describe", "--tags"); err == nil {
```
Comment (jasmingacic) [inline internal/build/build.go:34]: so this relies on git being installed and pulling this data from git. I wouldn't do that

**Review by jasmingacic — CHANGES_REQUESTED** [review]

File: internal/build/build.go (hunk 1 above)
Comment (mbana) [reply internal/build/build.go]: Resolved. Please see latest commit.

Please, mark as resolved, if you are happy with it.

Thanks.

**Comment by mbana** [comment]
> while at it could you just change go.mod to go 1.18

This is quite risky. Wouldn't it require anyone doing a `go install` of `kusk` to have `go1.18`?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -1,36 +1,24 @@
...
         run: go test -v ./...
-
-      - name: test-GoReleaser-build
-        uses: goreleaser/goreleaser-action@v2
-        with:
-          distribution: goreleaser
-          version: v1.6.3
-          args: release --rm-dist --skip-publish --skip-validate
-        env:
-          GITHUB_TOKEN: ${{ secrets.CI_BOT_TOKEN }}
```
[the remark this answers is not in the document]
Comment (mbana) [reply .github/workflows/go.yml:35]: No.

It was just used as a test run. It was drastically slowing down the builds so I removed it (I added it to begin with). So, before making a tag, one would run

```sh
$ make build-goreleaser
```

Then test out the build before pushing the tag.

The `.github/workflows/goreleaser.yml` is still present and I didn't modify that.

Please, mark as resolved, if you are happy with answer.

Thanks.

**Comment by mbana** [comment]
> I'd ditch go install as an install option

I can do it as a separate PR ... ? What do you think?

**Review by jasmingacic — APPROVED** [review]
Ship it!

**Review by kylehodgetts — APPROVED** [review]

File: cmd/version_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+	writer := bytes.NewBufferString("")
+	version := "_some-version_"
+	command := NewVersionCommand(writer, version)
+	command.Run(nil, []string{})
+
+	expected := `kusk version _some-version_
+https://github.com/kubeshop/kusk/releases/latest
+`
+	actual := writer.String()
```
[the remark this answers is not in the document]
Comment (mbana) [reply cmd/version_test.go:46]: Noted.

Possibly just checking it contains the version is enough. I don't mind either way. I just thought that checking the whole output equals what we expect to be a better approach.

@jasmingacic and @ypoplavs, what do you think?

[the remark this answers is not in the document]
Comment (jasmingacic) [reply cmd/version_test.go:46]: agreed. Just check if it contains the version

[the remark this answers is not in the document]
Comment (mbana) [reply cmd/version_test.go:43]: Resolved.

Code just does `assert.Contains(actual, version)`.

**Comment by mbana** [comment]
Tested with:

```sh
./kusk version; ./kusk --version; ./kusk -v             
kusk version 1.0.5-rc1
https://github.com/kubeshop/kusk/releases/tag/v1.0.5-rc1
kusk version 1.0.5-rc1
https://github.com/kubeshop/kusk/releases/tag/v1.0.5-rc1
kusk version 1.0.5-rc1
https://github.com/kubeshop/kusk/releases/tag/v1.0.5-rc1
```

Which seems right to me.

**Pull request merged**